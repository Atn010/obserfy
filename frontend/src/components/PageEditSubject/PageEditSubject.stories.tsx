import React, { FC } from "react"
import PageEditSubject from "./PageEditSubject"

export default {
  title: "Core|PageEditSubject",
  component: PageEditSubject,
  parameters: {
    componentSubtitle: "Just a simple PageEditSubject",
  },
}

export const Basic: FC = () => (
  <PageEditSubject areaId="asdfasdf" subjectId="asdfadsfa" />
)
