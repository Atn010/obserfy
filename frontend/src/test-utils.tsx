/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
