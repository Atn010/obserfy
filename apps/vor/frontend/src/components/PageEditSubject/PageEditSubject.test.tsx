import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./PageEditSubject.stories"

describe("PageEditSubject", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
