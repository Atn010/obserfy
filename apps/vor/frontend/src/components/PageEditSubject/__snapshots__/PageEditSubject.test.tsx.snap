// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageEditSubject should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-9 {
    font-size: 20px;
  }
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 640px;
  margin: auto;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 8px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 20rem;
  height: 3rem;
  margin-bottom: 32px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 8px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 100%;
  height: 6rem;
  margin-bottom: 16px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 8px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 8rem;
  height: 2rem;
  margin-bottom: 8px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 8px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 100%;
  height: 4rem;
  margin-bottom: 8px;
}

<div>
  <div
    class="emotion-9"
  >
    <div
      class="emotion-8"
    >
      <div
        class="emotion-0"
      />
      <div
        class="emotion-0"
      />
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      />
      <div
        class="emotion-4"
      />
      <div
        class="emotion-4"
      />
      <div
        class="emotion-4"
      />
      <div
        class="emotion-4"
      />
    </div>
  </div>
</div>
`;
